convert2rhel/pkghandler.py

```diff
@@ -1144,52 +1111,147 @@ def clean_yum_metadata():
+    evr2 = (version2_components[1], version2_components[2], version2_components[3])
+
+    return rpm.labelCompare(evr1, evr2)
+
+    # rename prev use of compare_package_versions to compare_package_strings
+
+
+def parse_pkg_string(pkg):
+    """
+    This function takes a version string in NEVRA, NEVR, NVRA, NVR, ENVRA, ENVR and decides whether
+    to parse with a yum/dnf module based on the package manager type of the system
+    """
+    if pkgmanager.TYPE == "yum":
+        pkg_ver_components = _parse_pkg_with_yum(pkg)
+    else:
+        pkg_ver_components = _parse_pkg_with_dnf(pkg)
+    return pkg_ver_components
+
 
-    # TODO(r0x0d): This function still needs some enhancements code-wise, it
-    # workes perfectly, but the way we are handling the versions is not 100%
-    # complete yet.  will be done in a future work. Right now, all the list of
-    # changes are listed in this comment:
-    # https://github.com/oamg/convert2rhel/pull/469#discussion_r873971400
-    pkg_ver_components = []
-    for pkg in pkg_1, pkg_2:
-        # Remove the package name and split the rest between epoch + version
-        # and release + arch
-        epoch_version, release_arch = pkg.rsplit("-", 2)[-2:]
-        # Separate the (optional) epoch from the version
-        epoch_version = epoch_version.split(":", 1)
-        if len(epoch_version) == 1:
-            epoch = "0"
-            version = epoch_version[0]
+def _parse_pkg_with_yum(pkg):
+    """
+    Parse verison string using yum and rpmUtils splitFilename
+    :return: Return a list containing name, epoch, version, release, arch
+    """
+
+    # package is in NEVRA/NEVR format
+    if re.findall(r"-\d:", pkg):
+        name, epoch_version, release_arch = pkg.rsplit("-", 2)
+        epoch, version = epoch_version.split(":", 1)
+        # if NEVRA format
+        if re.findall(r".*(x86_64|s390x|i86|ppc64le|aarch64)$", release_arch):
+            release, arch = release_arch.rsplit(".", 1)
+        # if NEVR format
         else:
-            epoch, version = epoch_version
-        # Discard the arch
-        release = release_arch.rsplit(".", 1)[0]
-        pkg_ver_components.append((epoch, version, release))
+            release = release_arch
+            arch = None
+
+    # package is in ENVR, ENVRA, NVR, NVRA
+    else:
+        (name, version, release, epoch, arch) = pkgmanager.splitFilename(pkg)
+
+        # splitFilename places part of the release in the arch field when no arch is present
+        # if arch field is invalid, append contents to release field
+        if arch not in ("x86_64", "s390x", "i86", "ppc64le", "aarch64"):
+            temp_release = arch
+            arch = None
+            release = release + "." + temp_release
+
+    pkg_ver_components = [name, epoch, version, release, arch]
+    # convert any empty strings to None for consistency
+    pkg_ver_components = [str(i or None) for i in pkg_ver_components]
```

Comment:
You could do that as well, without the need to shadow and override the variable. 

```suggestion
    # convert any empty strings to None for consistency
    pkg_ver_components = [str(i or None) for i in (name, epoch, version, release, arch)]
```

Reply:
Noted, i'll add that